pymc3/model.py

```diff
@@ -1929,15 +1936,7 @@ def Deterministic(name, var, model=None, dims=None):
     var = var.copy(model.name_for(name))
     model.deterministics.append(var)
     model.add_random_variable(var, dims)
-    var._repr_latex_ = functools.partial(_repr_deterministic_rv, var, formatting='latex')
-    var.__latex__ = var._repr_latex_
-
-    # simply assigning var.__str__ is not enough, since str() will default to the class-
-    # defined __str__ anyway; see https://stackoverflow.com/a/5918210/1692028
-    old_type = type(var)
-    new_type = type(old_type.__name__ + '_pymc3_Deterministic', (old_type,),
-        {'__str__': functools.partial(_repr_deterministic_rv, var, formatting='plain')})
-    var.__class__ = new_type
+    var.__class__ = DeterministicWrapper # adds str and latex functionality
```

Comment:
Is it possible for `var` to be a numpy array here, in which case, would assigning its `.__class__` attribute to a subclass of `tt.TensorVariable` cause problems? (I'm new to PyMC3 so sorry for the basic question :) )

Reply:
This will indeed cause errors if `var` is not a `TensorVariable`. I *think* a deterministic will always be a `TensorVariable`, at least it is in all the test scripts and demos. It's a good question, though; I'm not sure there is any formal specification requiring that it has to be. (But I cannot really see how a numpy-typed deterministic makes sense, given that all model factors are `TensorVariable`s anyway (and all results of e.g. numpy * tensorvariable etc. are also of type `TensorVariable`))